index-import.cpp

```diff
@@ -172,17 +177,21 @@ static IndexUnitWriter remapUnit(const std::unique_ptr<IndexUnitReader> &reader,
   return writer;
 }
 
-static int cloneRecord(StringRef from, StringRef to) {
+static bool cloneRecord(StringRef from, StringRef to) {
   // Two record files of the same name are guaranteed to have the same
   // contents, because the filename contains a hash of its contents. If the
   // destination record file already exists, no action needs to be taken.
   if (fs::exists(to)) {
-    return 0;
+    return true;
   }
 
   // With swift-5.1, fs::copy_file supports cloning. Until then, use copyfile.
-  return copyfile(from.str().c_str(), to.str().c_str(), nullptr,
-                  COPYFILE_CLONE);
+  int failed = copyfile(from.str().c_str(), to.str().c_str(), nullptr,
+                        COPYFILE_CLONE);
+
+  // In parallel mode we might be racing against other threads trying to create
+  // the same record. To handle this, just silently drop file exists errors.
```

Comment:
@kastiglione time to use [RxCpp](https://github.com/ReactiveX/RxCpp) to have proper parallel mode without racing ( ͡° ͜ʖ ͡°)

Reply:
Pull requests welcome 😂

Reply:
Laughs in SEGFAULT 
<img width="60" alt="angrylaughcry" src="https://user-images.githubusercontent.com/967132/63200840-93fbd580-c037-11e9-88bb-f8d1781d731e.png">